cinema/views.py

```diff
@@ -46,13 +95,77 @@ def get_serializer_class(self):
 
 class MovieSessionViewSet(viewsets.ModelViewSet):
     queryset = MovieSession.objects.all()
-    serializer_class = MovieSessionSerializer
 
-    def get_serializer_class(self):
+    def get_queryset(self) -> QuerySet:
+        self.queryset = self.queryset.select_related(
+            "movie",
+            "cinema_hall"
+        ).annotate(
+            tickets_available=(
+                F("cinema_hall__rows") * F("cinema_hall__seats_in_row")
+            ) - Count("tickets")
+        )
+
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+        if date:
+            try:
+                date = datetime.strptime(date, "%Y-%m-%d").date()
```

Comment:
The checklist advises not to split the date as it's already in the required format. Instead of parsing the date string, you can directly filter using the date string.